src/components/document/document-workspace.tsx

```diff
@@ -399,4 +415,13 @@ export class DocumentWorkspaceComponent extends BaseComponent<IProps> {
       return this.stores.documents.getDocument(documentKey);
     }
   }
+
+  private setExpandWorkspace = (expand: boolean) => {
+    this.setState({ expandWorkspace: expand });
+  }
+
+  private setExpandResources = (expand: boolean) => {
+    const { ui } = this.stores;
+    ui.toggleNavTabContent(expand);
+  }
```

Comment:
Ah, I see.  This is where some is in MobX and some is in React state.  Can we take this out of MobX and have it all happen here or are there other components that need access to `toggleNavTabContent` or the tab expanded state?
For consistency, I might call both of these functions toggle functions.  It's more in line with what the existing MobX function does and more indicative of behavior.  `setExpandWorkspace` could imply that it only turns expand on.  Maybe `toggleExpandWorkspace` and `toggleExpandResources`.  That's just my preference though.

Reply:
I realize I never commented on @mklewandowski's original point about MobX vs. React state. The fundamental question there is whether there are (or are likely to be) other clients that need to respond to changes in the UI state. If it's something that's truly local then React state is fine. If there are other clients that might need to know when the state changes, then it should be stored where that is possible, e.g. MobX (or React context). The other issue is that we aren't currently saving/restoring the UI state, but we probably should be, in which case whether or not it's something that should be saved/restored as part of the UI state is another consideration. I would think that the position of the divider probably should be saved/restored at some point, which is an argument for putting in MobX.